Plugins/Util/Util.cpp

```diff
@@ -258,4 +260,14 @@ ArgumentStack Util::EncodeStringForURL(ArgumentStack&& args)
     return stack;
 }
 
+ArgumentStack Util::GetTwoDARowCount(ArgumentStack&& args)
+{
+    ArgumentStack stack;
+    const auto twodaRef = Services::Events::ExtractArgument<std::string>(args);
+    auto *twoda = new C2DA(CResRef(twodaRef.c_str()), 0);
```

Comment:
This is leaking the memory. If you're using `new`, you need to use `delete` at the end. Or, better yet, you can use `unique_ptr<C2DA>` and it will automatically release it when the function exits (this will happen even if you exit due to an exception, etc)

Reply:
Or, better-better-yet, just use a local variable:
```C
C2DA twoda(CResRef(twodaRef.c_str()), 0);
twoda.Load2DArray();
... twoda.m_nNumRows...
````

Reply:
As per discord discussion, we used `Globals::Rules()->m_p2DArrays->GetCached2DA(twodaRef.c_str(), true);` as the most efficient means to retrieve the info.